src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -74,7 +76,7 @@
     public AddOperation(@JsonProperty("path") final JsonPointer path,
         @JsonProperty("value") final JsonNode value)
     {
-        super("add", path, value);
+        super(OPERATION_NAME, path, value);
```

Comment:
I wasn't aware of the serialization uses.  I'm actually pretty okay with
how it is now.

On Wed, Sep 23, 2015 at 8:13 AM, hnguyen08 <notifications@github.com> wrote:

> In src/main/java/com/github/fge/jsonpatch/AddOperation.java
> <https://github.com/box-metadata/json-patch/pull/2#discussion_r40215486>:
>
> > @@ -74,7 +76,7 @@
> >      public AddOperation(@JsonProperty("path") final JsonPointer path,
> >          @JsonProperty("value") final JsonNode value)
> >      {
> > -        super("add", path, value);
> > +        super(OPERATION_NAME, path, value);
>
> The way to resolve this is to have the factory pass the operation name
> into this constructor, so operation name is located only in the factory.
> However, there are still some places where we instantiate the operations
> directly with the constructor by passing in path & value rather than from
> JsonNode (e.g. MoveOperation instantiates a RemoveOperation and an
> AddOperation), so these are places where we cannot use the factory in its
> current form.
>
> In a future change, I think we can move instantiation by path & value into
> the factory itself, and highly discourage instantiation by the actual
> constructor. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/box-metadata/json-patch/pull/2/files#r40215486>.
>